### DISCUSSION TITLE
20241107 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#115] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - x 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust (14h)

#### Bittorrent Primitives

- [Updated licence](https://github.com/torrust/bittorrent-primitives/issues/1) to use the same one we are using in the [Bencode2Json](https://github.com/torrust/bencode2json) package.

#### Bencode2Json

- Published version [0.1.0](https://crates.io/crates/bencode2json)
- Someone on reddit gave me feedback and I opened an [issue](https://github.com/torrust/bencode2json/issues/7). There is no way to espace the `<hex>` tag we are using to identify byte sequences in the JSON. Therefore, there is no way to convert back to bencode deterministically. For example, if the encoded string is like this `11:<hex>ff</hex>` (valid UTF-8), you would get this JSON string `"<hex>ff</hex>"` that does not come from the encoded string `1:\xFF`.

#### Torrust Index

- https://github.com/torrust/torrust-index/pull/757

#### Tracker

- https://github.com/torrust/torrust-tracker/pull/1068
- https://github.com/torrust/torrust-tracker/issues/1067

4 months ago I was working on a new workflow to upload the code coverage to codecov:

https://github.com/josecelano-test/pull_request_target

There was an issue with the new version of the action (`codecov/codecov-action@v4`):

https://github.com/codecov/codecov-action/issues/1508

They have made some changes in the backend. I wanted to try again but there has also been a change in Rust and I can't generate the code coverage. The [Rust unstable flag `profile` was removed](https://github.com/rust-lang/compiler-team/issues/798). There is no update for the action we are using [alekitto/grcov@v0.2](https://github.com/alekitto/grcov). I'm trying to generate the report without an action:

https://github.com/torrust/torrust-tracker/issues/1075

The report looks fine locally and it's also uploaded to codecov on the [sample project](https://github.com/josecelano-test/pull_request_target) but codecov doesn't show any data. I'm [waiting for their feedback](https://github.com/codecov/codecov-action/issues/1508#issuecomment-2457969450) but I'm considering simpler alternatives like publishing the HTML report to GitHub Pages. The problem is we would not have the report for PRs.

#### Community Support

I'm helping a torrust user to install it on a VM.

**Comment by mario-nt** [comment]
### Index
- https://github.com/torrust/torrust-index/pull/759
- https://github.com/torrust/torrust-index/pull/764

### Other
- Renew expired GPG key
